pkg/provider/helm/values_test.go

```diff
@@ -313,6 +441,726 @@ func TestHelmValuesGenerator_GenerateValues_failure(t *testing.T) {
 	}
 }
 
+func TestGetNestedMap(t *testing.T) {
+	tests := []struct {
+		name   string
+		values map[string]any
+		key    string
+		want   map[string]any
+		exists bool
+	}{
+		{
+			name: "map exists, valid key",
+			values: map[string]any{
+				"spire-server": map[string]any{
+					"fullnameOverride": "spire-server",
+				},
+			},
+			key: "spire-server",
+			want: map[string]any{
+				"fullnameOverride": "spire-server",
+			},
+			exists: true,
+		},
+		{
+			name: "map doesn't exist, valid key",
+			values: map[string]any{
+				"spire-server": map[string]any{
+					"caKeyType": "rsa-2048",
+					"caTTL":     "12h",
+				},
+			},
+			key: "global",
+			want: map[string]any{
+				"spire": map[string]any{
+					"clusterName": "local1",
+				},
+			},
+			exists: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resp, exists := getNestedMap(tt.values, tt.key)
+
+			assert.Equal(t, tt.exists, exists)
+			if tt.exists {
+				assert.Equal(t, tt.want, resp)
+			}
+		})
+	}
+}
+
+func TestMergeMaps(t *testing.T) {
+	tests := []struct {
+		name                  string
+		src                   map[string]any
+		dest                  map[string]any
+		overwriteExistingKeys bool
+		want                  map[string]any
+	}{
+		{
```

Comment:
Is there a case where the types being merged don't align? Would be good to cover that case. e.g. a value in src is a map, but a string or slice in dest. and vice-versa.

Reply:
I've added a few [test cases](https://github.com/cofide/cofidectl/pull/62/commits/0b4c7e90492561f04c0d21da16bc3b4320202f42) for this. The current behaviour is that the `src` value will overwrite the existing `dest` value if their respective keys match, regardless of the type.